### ISSUE TITLE
Node version notices

### ISSUE THREAD

**Issue by jdalton** [greenkeeperio/greenkeeper#304]
With support for Node 0.10 ending today and 0.12 ending on Dec 31 (see [LTS docs](https://github.com/nodejs/LTS)) I noticed that there are plenty of 0.10 and 0.12 runs in folks `.travis.yml` configs.

_Heres a [Google BigQuery Github dataset](https://cloud.google.com/bigquery/public-data/github) query_:

<img width="152" src="https://cloud.githubusercontent.com/assets/4303/19011492/a280d736-874b-11e6-8f96-5fc8c95caf19.png">

Is there any interest in letting folks know via a `greenkeeper` PR that they're testing unsupported Node versions in their travis or circle ci configs?

\@thealphanerd

Related to #93.

Pinging in some [nodesecurity](https://github.com/orgs/nodesecurity) folks, @evilpacket,  @lukekarrys, @nlf, & @wraithgar as it may be something or interest for them too.

**Comment by jdalton** [comment]
Related to #93.

**Comment by jdalton** [comment]
Pinging in some [nodesecurity](https://github.com/orgs/nodesecurity) folks \@evilpacket  @lukekarrys @nlf @wraithgar. As it may be something or interest for them too.

**Comment by TheAlphaNerd** [comment]
The EOL of v0.10 has been pushed to October 31 and has been reflected on the LTS calendar

https://github.com/nodejs/LTS#lts-schedule

Let me know if there is anything I can help with in regards to planning

**Comment by boennemann** [comment]
Hey @jdalton and @TheAlphaNerd,

we're already syncing node versions in .travis.yml, package.json engines and .nvmrc files.
It would be great to send out a pull request that removes 0.10 for everyone on Nov 1st, and I think it should be doable for us.

I'd really appreciate if we could collect some ideas on how that could look like exactly.

For `.travis.yml`s we could just remove 0.10 from the versions. If it's the only one just add 4, or 6, or both? Should we remove 0.12 as well? Should we remove everything iojs?
Update the engines field to `>=4` unless already higher.
Update the `.nvmrc` to 6 (or 4?)

Would also be great to have some input for a pull request body that is insightful and convincing.

**Comment by jdalton** [comment]
@boennemann !!!!

Moving folks to 4 and 6 sounds good.

iojs is no longer a thing so upgrading would be awesome!
v5 is unsupported and should be upgraded to v6.

0.12 ends in Dec 31 so it may be easier to follow the LTS dates. It could also avoid hurt feels if someone feels their minimum bar is being dropped or forced to be dropped too early.

When choosing between 4 and 6 or any LTS and latest stable I'd treat them like PRs were you give folks both and they nix one. (kind of like how greenkeeper does PRs for major bumps just in case).
That said v6 will be LTS by mid Oct so by then folks could be directed to v6.

**Comment by jdalton** [comment]
@boennemann 

Other things to keep in mind is there's lots of versions within like `0.10.123`. To get the BigQuery data I normalized things a bit. I didn't have access to a yml parsing package but you get the idea:

SQL query:
<details>
```sql
SELECT 
  COUNT(*) as cnt, version
FROM 
  JS(
    (SELECT content FROM [bigquery-public-data:github_repos.contents] WHERE id IN (
      SELECT id FROM [bigquery-public-data:github_repos.files] WHERE RIGHT(path, 11) = ".travis.yml" 
    )),
    content,
    "[{ name: 'version', type: 'string'}]",
    "function(row, emit) {
       var yml = row.content || '';
       var match = yml.match(/^\s*node_js\s*:((?:\s*-.+)+|\s*\[[\s\S]+?\])/m);
       var snippet = match ? match[1] : '';
       var cleaned = snippet.replace(/#.+/g, '');
       var versions = cleaned.match(/[\w.&*]+/g) || [];

       versions.forEach(function(ver) {
         var match = ver.match(/^v?([0.]*\d+)/);
         var snippet = match ? match[1] : '';
         var cleaned = snippet.replace(/\b0{2,}\b/g, '0').replace(/^\./, '0.');
         var major = cleaned || ver.replace(/\bio\.js\b/g, 'iojs');

         emit({ version: major });
       });
     }" 
  )
GROUP BY version
ORDER BY cnt DESC
LIMIT 50
```
</details>

**Comment by jdalton** [comment]
Thank you @boennemann!

The PR is super nice!
https://github.com/search?q=Node.js+0.10+is+unmaintained&type=Issues

**Comment by jdalton** [comment]
> Pinging in some nodesecurity folks, ... as it may be something or interest for them too.

@evilpacket Any update or plans from the [NodeSecurity](https://github.com/nodesecurity) side of things?

**Comment by denis-sokolov** [comment]
If the bot suggest a developer to drop the support for old Node versions and modifies their `travis.yml`, we really need to make sure it updates the `engines` field or otherwise it will break their clients. In other words, given that this issue is implemented now, I think additional priority for #93 would be very valuable.
